pkg/querier/queryrange/roundtrip.go

```diff
@@ -88,8 +93,12 @@ func NewTripperware(cfg Config, log log.Logger, limits Limits, codec Codec, cach
 	if cfg.AlignQueriesWithStep {
 		queryRangeMiddleware = append(queryRangeMiddleware, InstrumentMiddleware("step_align"), StepAlignMiddleware)
 	}
-	if cfg.SplitQueriesByDay {
-		queryRangeMiddleware = append(queryRangeMiddleware, InstrumentMiddleware("split_by_day"), SplitByDayMiddleware(limits, codec))
+	// SplitQueriesByDay is deprecated use SplitQueriesByInterval.
+	if cfg.SplitQueriesByDay == true {
+		cfg.SplitQueriesByInterval = day
+	}
+	if cfg.SplitQueriesByInterval != 0 {
+		queryRangeMiddleware = append(queryRangeMiddleware, InstrumentMiddleware("split_by_day"), SplitByIntervalMiddleware(cfg.SplitQueriesByInterval, limits, codec))
```

Comment:
Do we wanna keep `split_by_day` for Prometheus histogram label and tracing span here? I believe `split_by_interval` would be clearer given the changes.